### PR TITLE
Add packages agent

### PR DESCRIPTION
**Pull request by itzg**
# Resolves

https://jira.rax.io/browse/SALUS-594

# What

Adds support for the packages agent (https://github.com/racker/salus-packages-agent/pull/1) and adds a new ingestor for influx line protocol.

# How

Some of the incremental changes were:
- refactorted the line protocol parsing into its own package so both agents and ingest packages could use it without encroaching on package cycles https://github.com/racker/salus-telemetry-envoy/commit/f2805d07e32b26d4889cfbec18998b1daecfb29b
- added a line protocol ingestor https://github.com/racker/salus-telemetry-envoy/commit/53b5de5c6957b76846a8f8d121daa6995c059c98
- refactored some functionality that is common to this agent and filebeat where they both use file based configs https://github.com/racker/salus-telemetry-envoy/commit/c5bd63624c3eb004fd7ec6afef1ce06aa8adf37e
- added the package agent runner https://github.com/racker/salus-telemetry-envoy/commit/4ae666da191cdd4529fb356e5f985571c1cc4a2c
- implemented test-monitor https://github.com/racker/salus-telemetry-envoy/pull/73/commits/e44c640982415f064dc3a52593d1a66f58c7f1f8

## How to test

Added unit tests

# TODO

- [x] implement `(pr *PackagesAgentRunner) ProcessTestMonitor`

### CODE REVIEW

File: agents/agents.go [hunk 1]
```diff
@@ -206,3 +210,111 @@ func handleContentConfigurationOp(op *telemetry_edge.ConfigurationOp, configInst
...
+	if err != nil {
+		log.WithError(err).WithField("path", configsPath).
+			Warn("unable to read files in configs directory")
+		return false
+	}
+
+	hasConfigs := false
+	for _, name := range names {
+		if path.Ext(name) == configFileExtension {
+			hasConfigs = true
```
Comment (GeorgeJahad) [inline agents/agents.go]: break?

File: agents/pkgagent_test.go [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+			{
+				Id:   "monitor-rpm",
+				Type: telemetry_edge.ConfigurationOp_REMOVE,
+			},
+		},
+	}
+	err = runner.ProcessConfig(configure)
+	require.NoError(t, err)
+
+	found, err := readFilesIntoMap(dataPath, ".json")
```
Comment (GeorgeJahad) [inline agents/pkgagent_test.go]: nit: maybe read the files in before the ProcessConfig() to confirm something really was there.

File: agents/pkgagent_test.go [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+		AgentType: telemetry_edge.AgentType_PACKAGES,
+		Operations: []*telemetry_edge.ConfigurationOp{
+			{
+				Id:       "monitor-rpm",
+				Type:     telemetry_edge.ConfigurationOp_CREATE,
+				Content:  `{"type":"packages", "includeRpm":true}`,
+				Interval: 3600,
+			}},
+	}
+	// assumes testing from previous test case
```
Comment (GeorgeJahad) [inline agents/pkgagent_test.go]: i don't understand this assumption

File: agents/agents.go (hunk 1 above)
Comment (itzg) [reply agents/agents.go]: Ah yes

File: agents/pkgagent_test.go (hunk 2 above)
Comment (itzg) [reply agents/pkgagent_test.go]: Will do

File: agents/pkgagent_test.go (hunk 3 above)
Comment (itzg) [reply agents/pkgagent_test.go]: I'm going to change this to
```
	// it's safe to the real ProcessConfig since it was tested in previous test case
```

**Comment by itzg** [comment]
Just to note, for integration testing (and sprint demo) on my Mac, I created a fake `dpkg-query` script and placed it in my `$PATH`:

```
#!/bin/sh

cat <<EOF
adduser 3.118 all
apt 1.8.2 amd64
base-files 10.3+deb10u2 amd64
base-passwd 3.5.46 amd64
bash 5.0-4 amd64
bsdutils 1:2.33.1-0.1 amd64
coreutils 8.30-3 amd64
dash 0.5.10.2-5 amd64
debconf 1.5.71 all
debian-archive-keyring 2019.1 all
debianutils 4.8.6.1 amd64
diffutils 1:3.7-3 amd64
dpkg 1.19.7 amd64
e2fsprogs 1.44.5-1+deb10u2 amd64
fdisk 2.33.1-0.1 amd64
findutils 4.6.0+git+20190209-2 amd64
gcc-8-base 8.3.0-6 amd64
gpgv 2.2.12-1+deb10u1 amd64
grep 3.3-1 amd64
gzip 1.9-3 amd64
EOF
```

**Comment by itzg** [comment]
@jjbuchan @harageth as with the other PRs, I'm going to merge this to close out the story for the sprint, but feel free to post comments and I'll open new PRs as needed.

**Pull request merged**